source/module_hsolver/genelpa/elpa_new_real.cpp

```diff
@@ -248,7 +248,7 @@ int ELPA_Solver::decomposeRightMatrix(double* B, double* EigenValue, double* Eig
             {
                 int iGlobal = globalIndex(i, nblk, nprows, myprow);
                 if (iGlobal > jGlobal)
-                    B[i + j * narows] = 0;
+                    B[i + j * static_cast<std::size_t>(narows)] = 0;
```

Comment:
This loop might cause pool performance.

Reply:
> This loop might cause pool performance.

You are right.
It's better to move both `j * narows` and `j * static_cast<std::size_t>(narows)` out of loop `i`.
I just consider this is an optimization, which shouldn't be included in this commit.